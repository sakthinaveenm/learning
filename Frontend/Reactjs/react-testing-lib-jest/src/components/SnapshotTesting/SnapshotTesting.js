import React,{useState} from 'react'

const STATUS = {
    HOVERED : 'hovered',
    NORMAL : 'normal'
}

function SnapshotTesting({page,children}) {
const [status,setStatus] = useState(STATUS.NORMAL);

const onMouseOver =()=>{
    setStatus(STATUS.HOVERED)
}
const onMouseLeave =()=>{
    setStatus(STATUS.NORMAL)
}
  return (
    <div>
        <h1>
        SnapshotTesting
        </h1>
        <a 
        className={status}
        onMouseOver={onMouseOver}
        onMouseLeave={onMouseLeave}
        href={page || ''}
        >
            {children === undefined || children === null || children !== '' ? 'No Data' : children}
        </a>
        </div>
  )
}

export default SnapshotTesting