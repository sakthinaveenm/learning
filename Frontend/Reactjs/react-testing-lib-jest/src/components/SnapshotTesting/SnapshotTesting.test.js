import SnapshotTesting from "./SnapshotTesting";
import renderer from 'react-test-renderer';


it('changes class when hovered',()=>{
    const component = renderer.create(
        <SnapshotTesting page="">Children data</SnapshotTesting>
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(()=>{
    console.log(tree.props);
        tree.props.onMouseOver();

    })
    // tree.props.onMouseEnter()

})