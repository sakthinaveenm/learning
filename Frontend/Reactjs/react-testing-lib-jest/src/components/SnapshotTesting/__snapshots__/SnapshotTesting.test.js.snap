// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes class when hovered 1`] = `
<div>
  <h1>
    SnapshotTesting
  </h1>
  <a
    className="normal"
    href=""
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    No Data
  </a>
</div>
`;
